xchange-stream-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingService.java

```diff
@@ -163,6 +163,19 @@ public String getSubscribeMessage(String channelName, Object... args) throws IOE
       }
       subscriptionRequestMap.put(reqID, channelName);
 
+      Observable.timer(2, TimeUnit.SECONDS)
```

Comment:
I'm a little concerned about what happens if we either close down or remove the subscription between sending a subscription and this timer firing.  The latter probably isn't a big issue, but the former will result in some very odd errors.

Can we add something in here to skip the error if we've been shut down when the observable fires, or if the channel has been removed?

Reply:
If the connection is closed, then the `.takeWhile(t -> isSocketOpen())` would not pass and the error will not be generated.

If the consumer unsubscribed (i.e. disposed of the Disposable), then the `onError` handled will not be called at all.

I added a few tests to illustrate this behaviour. I think no odd errors are logged.
I may have missed a scenario, though. Let me know if you think of some.